print-domain.c

```diff
@@ -267,6 +267,181 @@ ns_cprint(netdissect_options *ndo,
 	return (cp + i);
 }
 
+static void
+print_eopt_ecs(netdissect_options *ndo, const u_char *cp,
+               u_int data_len)
+{
+    u_int family, addr_bits, src_len, scope_len;
+
+    u_char padded[32];
+    char addr[INET6_ADDRSTRLEN];
+
+    if (!ND_TTEST_2(cp)) {
```

Comment:
Is this test for truncated condition?
If yes, it is now useless because the new GET_ macros have embedded bounds checks, via get_ inline functions (see extract.h).

Reply:
The intent of the checks was to ensure that the EDNS option was properly formatted (e.g. ECS should have 2 bytes for family, 1 byte for source length, etc.). So I was marking the option invalid if it didn't have the proper number of bytes.

I don't know if there's a way to differentiate between a malformed option and a truncated message in this case. If you have any suggestions on that, let me know.

Reply:
> The intent of the checks was to ensure that the EDNS option was properly formatted (e.g. ECS should have 2 bytes for family, 1 byte for source length, etc.).

I.e., you're checking to make sure that the *option length* is large enough?

That should be done by checking the value of `data_len`.

That's not what `ND_TTEST_2()` does.  It checks whether there are at least two bytes in *the entire packet, as captured*, starting at the byte pointed to by `cp`; by default, that checks against the end of the captured part of the link-layer frame, although, as DNS runs atop UDP or TCP, which run atop IP, it will check against the end of the captured part of the IP datagram.  It doesn't check the remaining length of the option data; that'd be done by checking `data_len`.
 
> I don't know if there's a way to differentiate between a malformed option and a truncated message in this case. If you have any suggestions on that, let me know.

Check against `data_len` and print a message *different* from the one printed if the bounds checks embedded in the GET_ macros fail.  Calling `nd_print_invalid()` will do that, but you could also print "option too short" or something such as that.

Reply:
Someting like :
```
    if (data_len < @minimal_length@) {
        ND_PRINT("[length %u < @minimal_lengtht@]", data_len);
        nd_print_invalid(ndo);
        return;
    }
```
is what you want to do?

Reply:
Yeah that looks better. I'll make that change in a minute.

There's also two `ND_TTEST` checks in `eopt_print()` (`print-domain.c:344`). While those are redundant with the new `GET_` would it be best to leave those since the general flow is to pass `NULL` up to `domain_print()` which handles truncation there?